src/Receipt/DonationReceipt.php

```diff
@@ -0,0 +1,439 @@
+<?php
+namespace Give\Receipt;
+
+use InvalidArgumentException;
+use function give_get_payment_meta as getDonationMetaData;
+use function give_get_gateway_admin_label as getGatewayLabel;
+use function give_get_donation_donor_email as getDonationDonorEmail;
+use function give_get_donation_address as getDonationDonorAddress;
+use function give_format_amount as formatAmount;
+use function give_currency_filter as filterCurrency;
+
+class DonationReceipt extends Receipt {
+	/**
+	 * Iterator initial position.
+	 *
+	 * @var int
+	 */
+	private $position = 0;
+
+	/**
+	 * Array of section ids to use for Iterator.
+	 * Note: this property helps to iterate over associative array.
+	 *
+	 * @var int
+	 */
+	private $sectionIds = [];
+
+	/**
+	 * Receipt donor section id.
+	 */
+	const DONORSECTIONID = 'Donor';
+
+	/**
+	 * Receipt donation section id.
+	 */
+	const DONATIONSECTIONID = 'Donation';
+
+	/**
+	 * Receipt additional information section id.
+	 */
+	const ADDITIONALINFORMATIONSECTIONID = 'AdditionalInformation';
+
+	/**
+	 * Donation id.
+	 *
+	 * @since 2.7.0
+	 * @var int $donationId
+	 */
+	public $donationId;
+
+	/**
+	 * Receipt constructor.
+	 *
+	 * @since 2.7.0
+	 * @param $donationId
+	 */
+	public function __construct( $donationId ) {
+		$this->donationId = $donationId;
+
+		$this->addDonorSection();
+		$this->addDonationSection();
+		$this->addSection( $this->getAdditionInformationSection() ); // Additional Information Section
+	}
+
+	/**
+	 * Add receipt section.
+	 *
+	 * @param  array  $section
+	 * @param  string $position Position can be set either "before" or "after" to insert section at specific position.
+	 * @param  string $sectionId
+	 *
+	 * @return Section
+	 *
+	 * @since 2.7.0
+	 */
+	public function addSection( $section, $position = '', $sectionId = '' ) {
```

Comment:
This isn't specific to donations, so you can just move it into the parent `Receipt` class. Same thing with the validation class.

Reply:
fixed: https://github.com/impress-org/givewp/compare/ccb59a12b626...55c57984a9f5